inst/include/Rcpp/r_cast.h

```diff
@@ -67,9 +67,17 @@ namespace Rcpp {
             default:
                 const char* fmt = "Not compatible with requested type: "
                                   "[type=%s; target=%s].";
+#ifndef NDEBUG
+                fprintf(stderr,
```

Comment:
I wasn't aware of `REprintf`, made the change.

I used the following to test:
```
Rcpp::cppFunction('void type_err(IntegerVector iv) {}')
type_err("a")
```
Which aborts with the appropriate error message if I add `-UNDEBUG` in `CXXFLAGS` before compilation, otherwise gives normal Rcpp behavior.

Reply:
Thanks.  `REprintf()` is one of those gems hidden away in WRE which "they" to nag you on submission.  You know, like 'no stdout, no abort, ...'  ;-)